valohai_yaml/objs/config.py

```diff
@@ -133,6 +134,15 @@ def get_step_by(self, **kwargs) -> Optional[Step]:
                 return step
         return None
 
+    def merge_with(self, other):
```

Comment:
What about endpoints and pipelines? This should not just silently leave them unmerged, IMHO.